feat(avt): a11y status page and a11y status table component

alisonjoseph opened the pull request:
Closes #3851
Closes #3848
Closes #3849

Building off work started in https://github.com/carbon-design-system/carbon-website/pull/3735 (should be able to close that now)

### Changelog

- New Accessibility status page `/components/overview/accessibility-status`
  - Note: added "not available" to screen reader tag list 
  - Update nav link
  - Add redirect to new url
- Update structure of data inside components.json
  - Add screen reader testing data
  - Add parent component data (to link to docs pages)
  - Add all components from testing data 
  - Add overview data, to allow us to list more components that don't appear with thumbnails on overview page (would we want a 1:1 match with thumbnails at some point?)
 - New `A11yStatus` component
 - New `A11ySTatusTag` component
 - Add `A11yStatus` component to all component a11y pages

### Testing
New page
- https://carbondesignsystem-git-a11y-status-carbon-design-system.vercel.app/components/overview/accessibility-status

Component filtered on individual component a11y pages
- https://carbondesignsystem-git-a11y-status-carbon-design-system.vercel.app/components/accordion/accessibility#accessibility-testing-status
- https://carbondesignsystem-git-a11y-status-carbon-design-system.vercel.app/components/dropdown/accessibility#accessibility-testing-status

alisonjoseph commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+import { Tag } from '@carbon/react';
+import A11yStatus from 'components/A11yStatus';
+
+<PageDescription>
+
+Components are one of the key building blocks of the design system. Their
+systematic reuse helps to create visual and functional consistency across
+products.
+
+</PageDescription>
```
> [!WARNING]  
> Need copy for the intro paragraph

tay1orjones replied:
My take on this:

```suggestion
<PageDescription>

Accessibility verification testing (AVT) involves the use of both manual and automated tools and software to assess the accessibility of Carbon components against established accessibility standards, such as the Web Content Accessibility Guidelines (WCAG).

Carbon implements a robust matrix of AVT to ensure that every Carbon component meets the utmost standards of accessibility. The status of this testing is outlined across the website through a detailed matrix of test categories and tags.

</PageDescription>
```

alisonjoseph replied:
This feels a bit long, when most pages are ~2 sentences for their intro text.
<img width="1206" alt="Screenshot 2024-01-05 at 2 14 10 PM" src="https://github.com/carbon-design-system/carbon-website/assets/2753488/4305c41c-ada6-48c4-be8f-e7dad8c317a2">

What about this? Maybe still too wordy?

```suggestion
<PageDescription>

Accessibility verification testing (AVT) involves the use of both manual and
automated tools and software to assess the accessibility of Carbon components
against established accessibility standards, such as the Web Content
Accessibility Guidelines (WCAG). The status of this testing is outlined across
the website through a detailed matrix of test categories and tags.

</PageDescription>
```

aubrey-oneal replied:
The page description does need to be short and I noticed both of your options had information that  was similar to the Accessibility tag status content. So here is an option where I massaged some of your messaging underneath that header:

Page description:
Manual and automated tests assess the accessibility of Carbon components against established standards. Read on to learn about how components are tagged with test statuses across the website.

Accessibility tag status

Automated accessibility verification tests (AVT) are run for each change proposed to the Carbon codebase. Additional manual tests are done periodically, such as screenreader testing. These tests ensure a consistent baseline of accessibility for each stable component.

Carbon implements a robust matrix of AVT to ensure that every Carbon component meets the utmost standards of accessibility. The status of this testing is outlined across the website through a detailed matrix of test categories and tags, outlined below.

alisonjoseph replied:
Thanks!

alisonjoseph commented on src/components/A11yStatus/a11y-status.module.scss:
```diff
@@ -0,0 +1,13 @@
+  white-space: nowrap;
+
+  span {
+    //override to fix alignment issue with variable font
```
TODO: check if we have latest version. alignment issues should have been fixed https://github.com/IBM/plex/issues/377

tay1orjones replied:
I can't remember how we're bringing in Plex, but the latest version binary is too large to update. We might need to pull it in via Akamai instead for the time being.

alisonjoseph replied:
Looks like they are being pulled directly from the theme.
https://github.com/carbon-design-system/gatsby-theme-carbon/tree/main/fonts
https://github.com/carbon-design-system/gatsby-theme-carbon/blob/main/packages/gatsby-theme-carbon/src/styles/internal/_plex.scss

We could probably update that to pull from Akamai instead? I'll take a look next week.

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+For every latest release, Carbon runs tests on all components to meet the
+accessibility requirements. These different statuses report the work that Carbon
+has done in the back end. These tests appear only when the components are
+stable.
```
My take at this

```suggestion
Automated accessibility verification tests are ran for each change proposed to the Carbon codebase. Additional manual tests are done periodically, such as screenreader testing. These tests ensure a consistent baseline of accessibility for each stable component. 

These accessibility tests can differ in their level of depth, breadth, and fidelity. A series of tags surface these differences throughout the website. The tags represent different levels of code test fidelity across individual components and types of testing within the system.
```

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+
+| Tag                                     | Meaning                                                                                                                                                                                                             |
+| --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| <Tag type="green">Tested</Tag>          | When all tests pass and there are no violations when running the automated Equal Access Checker on default state, advanced states, and keyboard navigation.                                                         |
```
```suggestion
| <Tag type="green">Tested</Tag>          | Tests have been implemented, pass, and show no violations via the [IBM Equal Access Accessibility Checker](https://github.com/IBMa/equal-access) for a components' default state, advanced states, and keyboard navigation.                                                         |
```

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+| Tag                                     | Meaning                                                                                                                                                                                                             |
+| --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| <Tag type="green">Tested</Tag>          | When all tests pass and there are no violations when running the automated Equal Access Checker on default state, advanced states, and keyboard navigation.                                                         |
+| <Tag type="blue">Partially tested</Tag> | When some tests pass or are removed, they are either combined into existing tests or substituted with new ones and moved to a different file. Skipped tests stay in the file but are clearly marked with test.skip. |
```
```suggestion
| <Tag type="blue">Partially tested</Tag> | Some testing has been done, but is still incomplete and in progress. For example, manual tests may not have covered all use cases, or automated tests may not cover all component states. Previously existing tests may have become invalid and are temporarily skipped or removed. |
```

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+| --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| <Tag type="green">Tested</Tag>          | When all tests pass and there are no violations when running the automated Equal Access Checker on default state, advanced states, and keyboard navigation.                                                         |
+| <Tag type="blue">Partially tested</Tag> | When some tests pass or are removed, they are either combined into existing tests or substituted with new ones and moved to a different file. Skipped tests stay in the file but are clearly marked with test.skip. |
+| <Tag type="teal">Manually tested</Tag>  | When screen reader tests are conducted manually and validated by a human to ensure quality.                                                                                                                         |
```
Manual testing has been performed and validated by a human to ensure proper accessibility, such as screen reader testing.

tay1orjones replied:
```suggestion
| <Tag type="teal">Manually tested</Tag>  | Manual testing has been performed and validated by a human to ensure proper accessibility, such as screen reader testing.                                                                                                                         |
```

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+| <Tag type="green">Tested</Tag>          | When all tests pass and there are no violations when running the automated Equal Access Checker on default state, advanced states, and keyboard navigation.                                                         |
+| <Tag type="blue">Partially tested</Tag> | When some tests pass or are removed, they are either combined into existing tests or substituted with new ones and moved to a different file. Skipped tests stay in the file but are clearly marked with test.skip. |
+| <Tag type="teal">Manually tested</Tag>  | When screen reader tests are conducted manually and validated by a human to ensure quality.                                                                                                                         |
+| <Tag>Not available</Tag>                | Usually, with keyboard navigation tests or screen reader, when this type of test is not available for certain components.                                                                                           |
```
```suggestion
| <Tag>Not available</Tag>                | In some cases, testing data is not available for certain components or component states. For instance, keyboard navigation testing is not necessary for components that are non-interactive.                                                                                           |
```

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+| <Tag type="blue">Partially tested</Tag> | When some tests pass or are removed, they are either combined into existing tests or substituted with new ones and moved to a different file. Skipped tests stay in the file but are clearly marked with test.skip. |
+| <Tag type="teal">Manually tested</Tag>  | When screen reader tests are conducted manually and validated by a human to ensure quality.                                                                                                                         |
+| <Tag>Not available</Tag>                | Usually, with keyboard navigation tests or screen reader, when this type of test is not available for certain components.                                                                                           |
+| <Tag type="purple">Not yet tested</Tag> | Usually, with experimental or unstable components, tests have not been conducted for either automated or manual testing.                                                                                            |
```
```suggestion
| <Tag type="purple">Not yet tested</Tag> | Manual and/or automated testing has been deferred. Most often this applies to experimental or unstable components that do not have testing performed until they are [moved to stable](https://github.com/carbon-design-system/carbon/blob/main/docs/experimental-code.md#moving-to-stable).                                                                                            |
```

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+| <Tag>Not available</Tag>                | Usually, with keyboard navigation tests or screen reader, when this type of test is not available for certain components.                                                                                           |
+| <Tag type="purple">Not yet tested</Tag> | Usually, with experimental or unstable components, tests have not been conducted for either automated or manual testing.                                                                                            |
+
+## Accessibility tests
```
```suggestion
## Accessibility test categories
```

alisonjoseph commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,67 @@
+<AnchorLinks>
+
+<AnchorLink>Accessibility tag status</AnchorLink>
+<AnchorLink>Accessibility tests</AnchorLink>
```
```suggestion
<AnchorLink>Accessibility test categories</AnchorLink>
```

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+
+## Accessibility tests
+
+Accessibility AVT (Accessibility Automated Verification Testing) involves the
+use of automated tools and software to assess the accessibility of a digital
+product against established accessibility standards, such as the Web Content
+Accessibility Guidelines (WCAG).
```
Every accessibility test falls into one of four categories. Marked with the tags above, these categories show the status for each of these discrete areas of testing. This additional detail helps clarify exactly what ways a component has been tested for accessibility.

tay1orjones replied:
```suggestion
Every accessibility test falls into one of four categories. Marked with the tags above, these categories show the status for each of these discrete areas of testing. This additional detail helps clarify exactly what ways a component has been tested for accessibility.
```

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+
+## All component accessibility status
+
+This table reflects the latest accessibility testing of only the stable
+components in the React framework at the moment.
```
```suggestion
This table reflects the current AVT status of stable components within `@carbon/react`.
```

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+
+| Test                | Meaning                                                                                                                               | Possible Tags                                                                                                                                               |
+| ------------------- | ------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Default state       | The default state of the component                                                                                                    | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
```
```suggestion
| Default state       | Test(s) that ensure the initial render state of a component is accessible                                                                                                    | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
```

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+| Test                | Meaning                                                                                                                               | Possible Tags                                                                                                                                               |
+| ------------------- | ------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Default state       | The default state of the component                                                                                                    | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
+| Advanced states     | Other states rather than the default state of a component. These could be interactive states of a component or its multiple variants. | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
```
```suggestion
| Advanced states     | Tests that ensure additional states of the component are accessible. This could be interactive states of a component or its multiple variants. | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
```

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+| ------------------- | ------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Default state       | The default state of the component                                                                                                    | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
+| Advanced states     | Other states rather than the default state of a component. These could be interactive states of a component or its multiple variants. | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
+| Keyboard navigation | This test ensures every action a mouse user can do should have a keyboard equivalent.                                                 | <Tag type="green">Tested</Tag><br/><Tag>Not available</Tag><br/><Tag type="blue">Partially tested</Tag>                                                     |
```
```suggestion
| Keyboard navigation | Tests that ensure focus is properly managed, and all interactive functions of a component have a proper keyboard-accessible equivalent.                                                 | <Tag type="green">Tested</Tag><br/><Tag>Not available</Tag><br/><Tag type="blue">Partially tested</Tag>                                                     |
```

tay1orjones commented on src/pages/components/overview/accessibility-status.mdx:
```diff
@@ -0,0 +1,62 @@
+| Default state       | The default state of the component                                                                                                    | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
+| Advanced states     | Other states rather than the default state of a component. These could be interactive states of a component or its multiple variants. | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
+| Keyboard navigation | This test ensures every action a mouse user can do should have a keyboard equivalent.                                                 | <Tag type="green">Tested</Tag><br/><Tag>Not available</Tag><br/><Tag type="blue">Partially tested</Tag>                                                     |
+| Screen reader       | This test ensures that the visual information on the screen is reinforced programmatically.                                           | <Tag>Not available</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="teal">Manually tested</Tag><br/><Tag type="blue">Partially tested</Tag> |
```
```suggestion
| Screen reader       | This manual testing ensures that the visual information on the screen is properly conveyed and read correctly by screen readers such as JAWS, VoiceOver, and NVDA.                                           | <Tag>Not available</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="teal">Manually tested</Tag><br/><Tag type="blue">Partially tested</Tag> |
```

tay1orjones reviewed (commented):
Sorry for hijacking this a bit with my suggestions 😅  

I figured it might be best to just slot in here for the content work I needed to do in https://github.com/carbon-design-system/carbon/issues/15326

@thyhmdo could you take a look here and flag anything you think doesn't sound right?

alisonjoseph commented on src/data/components.json:
```diff
@@ -1,124 +1,513 @@
+      "parentComponent": "Textinput",
+      "overview": false,
+      "testing": {
+        "screenreader": "manual"
```
@andreancardona have our fluid components been screen reader tested?

andreancardona replied:
unstable ones no. stable fluid yes!

alisonjoseph replied:
Thanks, updated the data to reflect that. I believe only fluid textinput is stable from what I can see.

mbgower commented:
This is a design question, not implementation question, so feel free to get me to repost elsewhere, but reviewing this, I had the following question:
Do we really need "Manually tested?

First, it is the longest tag, so it's much easier to design without it. Second, it implies that no keyboard testing is manually done, which I don't think is accurate. Third, I don't see what we lose by just having the 'pass' result of screen reader be "tested".
<img width="759" alt="image" src="https://github.com/carbon-design-system/carbon-website/assets/14298245/e8d8e17f-63a6-45b4-95c8-b6811b28ee99">

I suspect this has already been raised, and there is a good response, but I was struck by it during my review of the work. Thanks!
BTW, if this suggestion was adopted, there would have to be some slight wording in the meaning column:
> Tests have been implemented, pass, and show no violations, either via the [IBM Equal Access Accessibility Checker](https://github.com/IBMa/equal-access) for a components’ default state, advanced states, or via manual testing.

<img width="796" alt="image" src="https://github.com/carbon-design-system/carbon-website/assets/14298245/b3de1dc3-042e-4638-8134-2ef4a95df5d2">

thyhmdo commented:
> This is a design question, not implementation question, so feel free to get me to repost elsewhere, but reviewing this, I had the following question: Do we really need "Manually tested?
> 
> First, it is the longest tag, so it's much easier to design without it. Second, it implies that no keyboard testing is manually done, which I don't think is accurate. Third, I don't see what we lose by just having the 'pass' result of screen reader be "tested". <img alt="image" width="759" src="https://private-user-images.githubusercontent.com/14298245/295334776-e8d8e17f-63a6-45b4-95c8-b6811b28ee99.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4OTY1MDIsIm5iZiI6MTcwNDg5NjIwMiwicGF0aCI6Ii8xNDI5ODI0NS8yOTUzMzQ3NzYtZThkOGUxN2YtNjNhNi00NWI0LTk1YzgtYjY4MTFiMjhlZTk5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEwVDE0MTY0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxMjY0YjdmZjQ2Y2I0OGIwYzVkOTJjODhhZjg0MGU0M2RhNDhkNWJiZGVhNTNjZWNjNTAxZmVlMTBmNWQxOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.E2vrbq1eox2DoRElyPSUGTs7y-JlBeHX3cWAUlvaB8I">
> 
> I suspect this has already been raised, and there is a good response, but I was struck by it during my review of the work. Thanks! BTW, if this suggestion was adopted, there would have to be some slight wording in the meaning column:
> 
> > Tests have been implemented, pass, and show no violations, either via the [IBM Equal Access Accessibility Checker](https://github.com/IBMa/equal-access) for a components’ default state, advanced states, or via manual testing.
> 
> BTW, on reflection, since you break each component down by its default and advanced states, keyboard, and screen reader, there's no need to state these so thoroughly in the meaning, IMO. I also think a bit of the wording at the front is unnecessary, so I suggest:
> 
> > **Tests have been carried out, via either the [IBM Equal Access Accessibility Checker](https://github.com/IBMa/equal-access) or manual testing, and show no violations.**
> 
> <img alt="image" width="796" src="https://private-user-images.githubusercontent.com/14298245/295335159-b3de1dc3-042e-4638-8134-2ef4a95df5d2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4OTY1MDIsIm5iZiI6MTcwNDg5NjIwMiwicGF0aCI6Ii8xNDI5ODI0NS8yOTUzMzUxNTktYjNkZTFkYzMtMDQyZS00NjM4LTgxMzQtMmVmNGE5NWRmNWQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEwVDE0MTY0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFiMzRhZDBiNDgyOGIwY2U1OGFmZTM4ZWUwZDYwZTdjMWNkYzQ4YzQwY2Q3MTI0NWUzNzVhNGQyM2U4OTQ1YTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.E7sEbBmpDvVHKJEk6AvdRyuOBMll0QXM1Vq8oCI5riA">

In general, we want to make it clear for non-technical users or those who are not familiar with this to understand that other tests are automatic while Screen reader requires a manual process so that's why it has Manually tested tag. 

And yes, we still need some help on that wording. That's a good suggestion! Thanks @mbgower

mbgower commented:
> other tests are automatic while Screen reader requires a manual process

Keyboard is also often (but not always?) going to be manual too, though, right? I think the existing generic wording of "Partially tested" supports making "Tested" also generic.

> we want to make it clear for non-technical users or those who are not familiar with this to understand

I think you can provide the kind of context you're talking about in the test categories table. I would argue that the "Manually tested" wording makes the results busier without really being clearer ; I'd argue it actually makes it a bit more confusing.
![image](https://github.com/carbon-design-system/carbon-website/assets/14298245/0ae63435-24de-45d5-9482-f8d71d9b91b4)

thyhmdo commented:
Looks great! Here are a few things I have: 

1. The underline has a problem in the smaller breakpoint, under Keyboard navigation. 

![image](https://github.com/carbon-design-system/carbon-website/assets/77697351/d6983c10-3c58-4f01-be01-d8d80188c377)



2. For the text and the underline in the tag, can we somehow have them closer together with the text aligned more to the center? 

![image](https://github.com/carbon-design-system/carbon-website/assets/77697351/566bcc5f-4998-4fd0-8f39-a60508e115db)

3. I'd like to add a tooltip next to the "Accessibility testing status" on both Individual component page and the Overview page. I think we can improve the text but that's the idea. 

![image](https://github.com/carbon-design-system/carbon-website/assets/77697351/5043f658-de8b-4f79-b548-217b60389917)

4. I'm using Arc, and I know a lot more people are using this browser nowadays, the tag wrapping is still posing the issue. Is this something we can support as well? 

<img width="238" alt="image" src="https://github.com/carbon-design-system/carbon-website/assets/77697351/e98de26a-828f-48b1-9049-cc4791c44bca">

aubrey-oneal commented:
Hey @thyhmdo and @tay1orjones I spent some time drafting possible tooltips descriptions for your tags. Based on your designs, it seems this tooltip is always in the context of a component's state, so I did utilize that context. I also tried to clarify that 'Tested' implies automated testing—something that took me a little while to understand. I've _really_ paired down the info, probably more than I should. Please add back in what you need. My ideal character count for the tooltips would be 70 characters, 90 max. For Taylor's original 'meaning definitions' I only changed a word here or there.

**Tested**
- Meaning: Automated tests have been implemented, pass, and show no violations via the IBM Equal Access Accessibility Checker for a component’s default state, advanced states, and keyboard navigation.
- Tooltip (still too long): Passes automated tests and shows no violations via the IBM Equal Access Accessibility Checker. 

**Partially tested**
- Meaning: Some testing has been done, but is still incomplete and in progress. For example, manual tests may not have covered all use cases, or automated tests may not cover all component states. Previously existing tests may have become invalid and are temporarily skipped or removed.
- Tooltip: Some tests are incomplete, in progress, invalid, or temporarily skipped.

**Manually tested**
- Meaning: Manual testing has been performed and validated by a human to ensure proper accessibility, such as screen reader testing.
- Tooltip: A human has manually tested this component, e.g. screen reader testing.

**Not available**
- Meaning: In some cases, testing data is not available for certain components or component states. For instance, keyboard navigation testing is not necessary for components that are non-interactive.
- Tooltip: Test data is either not available or not applicable for this component state.

**Not yet tested**
- Meaning: Automated or manual testing has been deferred. Most often this applies to experimental or unstable components that do not have testing performed until they are moved to stable.
- Tooltip (still too long): Automated or manual testing has been deferred until this component state is stable.

tay1orjones commented:
> Keyboard is also often (but not always?) going to be manual too, though, right?

Once a component has been marked stable and had automated keyboard navigation tests written, it's actually the inverse because that testing is done on every pull request.

The tests simulate pressing tab and ensure the intended elements get focus in the proper order as documented on the a11y pages on the website for each component. Manual tests are done as well but it's less frequent/inconsistent.

tay1orjones commented:
> Keyboard is also often (but not always?) going to be manual too, though, right?

@mbgower Once a component has been marked stable and had automated keyboard navigation tests written, it's actually the inverse because that automated testing is done on every pull request.

The tests simulate pressing tab and ensure the intended elements get focus in the proper order as documented on the a11y pages on the website for each component. Manual tests are done as well but it's less frequent/inconsistent.

tay1orjones commented:
@aubrey-oneal Thank you! These look really great. Here's my take:

#### Tested
```diff
- Tooltip (still too long): Passes automated tests and shows no violations via the IBM Equal Access Accessibility Checker.
+ Tooltip: Passes all automated tests with no reported accessibility violations.
```

#### Not yet tested
```diff
- Tooltip (still too long): Automated or manual testing has been deferred until this component state is stable.
+ Tooltip: Automated or manual testing has been temporarily deferred.
```

mbgower commented:
> > Keyboard is also often (but not always?) going to be manual too, though, right?
> 
> @mbgower Once a component has been marked stable and had automated keyboard navigation tests written, it's actually the inverse because that automated testing is done on every pull request.
> 
> The tests simulate pressing tab and ensure the intended elements get focus in the proper order as documented on the a11y pages on the website for each component. Manual tests are done as well but it's less frequent/inconsistent.

Okay, but there still does not seem to be a lot of value having the "Manually tested" tag, especially given even with your qualifications, there can still be manual tests done for keyboard. Yet at no time do I see anyone using "manually tested" for keyboard.

In short, I feel like the wording suggestions I am proposing would reduce possible confusion, reduce the number of tags that can be used, and create a more consistent message.

tay1orjones commented:
@mbgower I think the primary reasons for having a `Manually tested` tag are:

1. Informing users and stakeholders we're taking the time to actually do manual verification of these things
2. It's a separate process with different requirements than automated testing (different screenreaders could be used (or not), quality of testing depends on the tester and their methods, not all components are manually tested at the same time, it's only partially done on every PR, etc.)
3. Manual testing is not done as often as automated testing, there is greater possibility for gaps in coverage or things to break between manual tests.
4. Manual testing does not produce the same artifacts that are available for automated tests (links to test source code, for example)


With this in mind, maybe `Tested` is flawed like you suggested earlier and should be renamed to `Automatically tested` or something.

mbgower commented:
Here is the information provided in the descriptions of testing
![image](https://github.com/carbon-design-system/carbon-website/assets/14298245/30e2cf84-7128-4025-b094-4d7d5226c1ef)

There is no reason we could not elaborate more fully on some of the points you make within the description of Screen Reader testing -- which already says "manually". You'll note that the keyboard description at least infers there is some manual testing in some situations. I still don't see _what_ value there is in slapping "Manually tested" as a tag. Also, I'd argue that if by SR tests, we mean testing for the appropriate name, role, value; more of those are caught by automated tests all the time. So we could even add some nuance to the discussion there.

I'm also not sure about "Not available" and the combinations that are used of it. I like the combination that is used for keyboard, and i think the same combination could be used for Screen Reader. The following description infers it could even take place with advanced states. By the same token, I'm not sure why "Not tested" isn't showing up for Keyboard.

> In some cases, testing data is not available for certain components or component states. For instance, keyboard navigation testing is not necessary for components that are non-interactive.

alisonjoseph commented:
@thyhmdo I've fixed all the wrapping issues and added the help icon. I skipped number 2, the text alignment. I have a [PR open in the gatsby theme](https://github.com/carbon-design-system/gatsby-theme-carbon/pull/1395) to update to the latest version of Plex(variable) which adjusts some of the alignment issues, and will hopefully fix that. 

Also as a FYI the link color is off in the toggletip, we have a couple open issues in core Carbon around this. 
https://github.com/carbon-design-system/carbon/issues/14732
https://github.com/carbon-design-system/carbon/issues/15204
![Screenshot 2024-01-16 at 2 11 24 PM](https://github.com/carbon-design-system/carbon-website/assets/2753488/79ccc7ce-e008-4099-b328-d0771e2ec95b)

@tay1orjones @aubrey-oneal I updated the copy for the descriptions.

thyhmdo commented:
Looking great!! Just a few minor things


---
This is on me, but I'm looking to change the text to:

> Learn more about tag and test meaning
> View all component accessibility status

![image](https://github.com/carbon-design-system/carbon-website/assets/77697351/2638985a-8d0d-4f92-b78e-611542f63a9c)

---
Change to with the lower case of "tags"

![image](https://github.com/carbon-design-system/carbon-website/assets/77697351/db043f53-9175-449c-8723-b14c46e054dc)


----
I noticed that this [one variant page](https://carbondesignsystem-git-a11y-status-carbon-design-system.vercel.app/components/accordion/accessibility#accessibility-testing-status) "View all component..." link takes me to the "Accessibility tag status" table instead of all the "All component accessibility status" section.              
While the [multiple variant page](https://carbondesignsystem-git-a11y-status-carbon-design-system.vercel.app/components/dropdown/accessibility#accessibility-testing-status) takes me to half of "Accessibility test categories" table?

alisonjoseph commented:
@thyhmdo just pushed a fix. The anchor links weren't working correctly because of the tooltip within the title.

thyhmdo reviewed (approved):
Looks great!

guidari commented:
@alisonjoseph One question.

When I try to access this component "Aspect Ratio" I get an 404. I know it is because we don't have a page in the website for the component so the error is totally right. But just make sure, should we cover that somehow or that's okay?
Might hard to filter these, idk...
[Link to access
](https://carbondesignsystem-git-a11y-status-carbon-design-system.vercel.app/components/overview/accessibility-status/#:~:text=Manually%20tested-,Aspect%20ratio,-Default%20state
)

alisonjoseph commented:
@guidari ah good catch, I'll fix that.

alisonjoseph commented:
fixed @andreancardona

andreancardona reviewed (approved):
🔥

mbgower commented:
Obviously kinda late for this comment, but I was a little surprised to see these appearing at the top of the accessibility tab.
Historically they've been near the bottom.  i guess I was focused on terminology and hadn't seen the intended location on the Accessibility tab in prior work -- or maybe I thought it was going to positioned in the right margin on the main page or something?

I did want to flag that it's a little odd that these are implemented with an H3, since the page structure is now wonky:
H3 Accessibility testing status
H2 What Carbon provides
H3 Keyboard interactions
...
H2 Development considerations

I don't think the status deserves an H2 -- its not on par with the other typical H2s:

- What Carbon provides
- Design recommendations
- Development considerations

IMO, it conceptually probably makes the most sense under Development considerations.

thyhmdo commented:
> @thyhmdo Obviously kinda late for this comment, but I was a little surprised to see these appearing at the top of the accessibility tab. Historically they've been near the bottom. i guess I was focused on terminology and hadn't seen the intended location on the Accessibility tab in prior work -- or maybe I thought it was going to positioned in the right margin on the main page or something?
> 
> I did want to flag that it's a little odd that these are implemented with an H3, since the page structure is now wonky: H3 Accessibility testing status H2 What Carbon provides H3 Keyboard interactions ... H2 Development considerations
> 
> I don't think the status deserves an H2 -- its not on par with the other typical H2s:
> 
> * What Carbon provides
> * Design recommendations
> * Development considerations
> 
> IMO, it conceptually probably makes the most sense under Development considerations.

Hey @mbgower thanks for the feedback and noted. We probably can change the order of it in the future. We also moved away from the margin space because of the condensed space, this will soon be reflected on tiles on the Usage tab work instead.
